qemu/tests/win_virtio_driver_installer_repair.py

```diff
@@ -91,9 +95,27 @@ def run(test, params, env):
         chk_cmd = params["vio_driver_chk_cmd"] % device_name[0:30]
         status = session.cmd_status(chk_cmd)
         if status != 0:
+            test.log.info("%s driver repair failed" % driver_name)
             unrepaired_driver.append(driver_name)
-    if unrepaired_driver:
-        test.fail("The missing %s driver repair failed."
-                  % unrepaired_driver)
+        else:
+            error_context.context("Run %s driver function test after repair"
+                                  % driver_name, test.log.info)
+            test_func = "win_driver_installer_test.%s_test" % test_name
+            driver_test_params = params.get('driver_test_params_%s'
+                                            % test_name, '{}')
+            driver_test_params = ast.literal_eval(driver_test_params)
+            if driver_name == 'viostor':
+                driver_test_params['images'] = 'stg0'
+            elif driver_name == 'vioscsi':
+                driver_test_params['images'] = 'stg1'
+            try:
+                eval("%s(test, params, vm, **driver_test_params)" % test_func)
+            except Exception:
+                fail_tests.append(test_name)
```

Comment:
You will need the exception details.
```suggestion
            except Exception as e:
                fail_tests.append('%s:\n%s' % (test_name, str(e)))
```

Reply:
Thanks @vivianQizhu . That will be more detailed.